wal/wal.go

```diff
@@ -252,6 +266,11 @@ func NewSize(logger log.Logger, reg prometheus.Registerer, dir string, segmentSi
 	return w, nil
 }
 
+// Compress returns if compression is enabled on this WAL.
+func (w *WAL) Compress() bool {
```

Comment:
Right now the `Checkpoint` method in `checkpoint.go` is using it to make the checkpoint WAL with the same compression configuration as the WAL the checkpoint is made from. I can refactor that function and Head to track the state but this was easier.

Reply:
Aah yeah , I thought Checkpoint is part of the wal struct so you you can access the w.compress variable directly.
I guess we can make Checkpoint() to be part of the WAL struct or keep it is as is if it makes more sense to keep it separately.

Maybe Cortex or Thanos need it separate?

cc @gouthamve , @bwplotka

Reply:
As of now Cortex does not use `Checkpoint()`. I am not as much of an expert in Thanos, but I believe they only ship Blocks not anything from the WAL.

Reply:
looking at https://github.com/prometheus/tsdb/pull/606 seems it will soon be wal.Checkpoint so I guess you might as well do it here to allow you to get rid of this exported method.

Reply:
Moving `checkpoint.go` into the `wal` package will cause an import cycle. Is it ok to briefly export the variable and hopefully as part of/after Callum's refactor we can get rid of it?

Reply:
sure